cmd/relay/main.go

```diff
@@ -0,0 +1,494 @@
+	"gorm.io/gorm"
+	"io"
+	"log/slog"
+	_ "net/http/pprof"
+	"net/url"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	libbgs "github.com/bluesky-social/indigo/cmd/relay/bgs"
+	"github.com/bluesky-social/indigo/cmd/relay/events"
+	"github.com/bluesky-social/indigo/cmd/relay/repomgr"
+	"github.com/bluesky-social/indigo/util"
+	"github.com/bluesky-social/indigo/util/cliutil"
+	"github.com/bluesky-social/indigo/xrpc"
+
+	_ "github.com/joho/godotenv/autoload"
+	_ "go.uber.org/automaxprocs"
+
+	"github.com/carlmjohnson/versioninfo"
+	"github.com/urfave/cli/v2"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"gorm.io/plugin/opentelemetry/tracing"
+)
+
+func init() {
+	// control log level using, eg, GOLOG_LOG_LEVEL=debug
+	//logging.SetAllLoggers(logging.LevelDebug)
+}
+
+func main() {
+	if err := run(os.Args); err != nil {
+		slog.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+func run(args []string) error {
+
+	app := cli.App{
+		Name:    "relay",
+		Usage:   "atproto Relay daemon",
+		Version: versioninfo.Short(),
+	}
+
+	app.Flags = []cli.Flag{
+		&cli.BoolFlag{
+			Name: "jaeger",
+		},
+		&cli.StringFlag{
+			Name:    "db-url",
+			Usage:   "database connection string for BGS database",
+			Value:   "sqlite://./data/bigsky/bgs.sqlite",
+			EnvVars: []string{"DATABASE_URL"},
+		},
+		&cli.BoolFlag{
+			Name: "db-tracing",
+		},
+		&cli.StringFlag{
+			Name:    "plc-host",
+			Usage:   "method, hostname, and port of PLC registry",
+			Value:   "https://plc.directory",
+			EnvVars: []string{"ATP_PLC_HOST"},
+		},
+		&cli.BoolFlag{
+			Name:  "crawl-insecure-ws",
+			Usage: "when connecting to PDS instances, use ws:// instead of wss://",
+		},
+		&cli.StringFlag{
+			Name:    "api-listen",
+			Value:   ":2470",
+			EnvVars: []string{"RELAY_API_LISTEN"},
+		},
+		&cli.StringFlag{
+			Name:    "metrics-listen",
+			Value:   ":2471",
+			EnvVars: []string{"RELAY_METRICS_LISTEN", "BGS_METRICS_LISTEN"},
+		},
+		&cli.StringFlag{
+			Name:    "disk-persister-dir",
+			Usage:   "set directory for disk persister (implicitly enables disk persister)",
+			EnvVars: []string{"RELAY_PERSISTER_DIR"},
+		},
+		&cli.StringFlag{
+			Name:    "admin-key",
+			EnvVars: []string{"RELAY_ADMIN_KEY", "BGS_ADMIN_KEY"},
+		},
+		&cli.IntFlag{
+			Name:    "max-metadb-connections",
+			EnvVars: []string{"MAX_METADB_CONNECTIONS"},
+			Value:   40,
+		},
+		&cli.StringFlag{
+			Name:    "resolve-address",
+			EnvVars: []string{"RESOLVE_ADDRESS"},
+			Value:   "1.1.1.1:53",
+		},
+		&cli.BoolFlag{
+			Name:    "force-dns-udp",
+			EnvVars: []string{"FORCE_DNS_UDP"},
+		},
+		&cli.IntFlag{
+			Name:    "max-fetch-concurrency",
```

Comment:
not used (many of these options seem to be not used, i'm not going to check any more)